dashboard/modules/serve/serve_agent.py

```diff
@@ -251,51 +251,44 @@ async def put_all_applications(self, req: Request) -> Response:
         from ray._private.usage.usage_lib import TagKey, record_extra_usage_tag
 
         try:
-            config = ServeDeploySchema.parse_obj(await req.json())
+            config: ServeDeploySchema = ServeDeploySchema.parse_obj(await req.json())
         except ValidationError as e:
             return Response(
                 status=400,
                 text=repr(e),
             )
 
+        config_http_options = {
+            "host": config.http_options.host,
+            "port": config.http_options.port,
+            "root_path": config.http_options.root_path,
+            "request_timeout_s": config.http_options.request_timeout_s,
+        }
+
         async with self._controller_start_lock:
             client = await serve_start_async(
                 detached=True,
-                http_options={
-                    "host": config.http_options.host,
-                    "port": config.http_options.port,
-                    "root_path": config.http_options.root_path,
-                    "location": config.proxy_location,
-                },
+                http_options=dict(
+                    {"location": config.proxy_location}, **config_http_options
+                ),
             )
 
-        # Check HTTP Host
-        host_conflict = self.check_http_options(
-            "host", client.http_config.host, config.http_options.host
-        )
-        if host_conflict is not None:
-            return host_conflict
-
-        # Check HTTP Port
-        port_conflict = self.check_http_options(
-            "port", client.http_config.port, config.http_options.port
-        )
-        if port_conflict is not None:
-            return port_conflict
-
-        # Check HTTP root path
-        root_path_conflict = self.check_http_options(
-            "root path", client.http_config.root_path, config.http_options.root_path
-        )
-        if root_path_conflict is not None:
-            return root_path_conflict
+        for option in config_http_options:
+            conflict_response = self.check_http_options(
+                option,
+                getattr(client.http_config, option),
+                getattr(config.http_options, option),
+            )
+            if conflict_response:
+                return conflict_response
 
-        # Check HTTP location
-        location_conflict = self.check_http_options(
+        # Handle location conflict separately since proxy_location isn't stored
+        # inside config.http_options
```

Comment:
Non-blocker: If we just store line 271 as `new_http_configs` or something, then we can check all of them in the same loop :)